### DISCUSSION TITLE
Editing and adding new properties to existing elements from the frontend API

### DISCUSSION THREAD

**Discussion by JohnMajerle** [iTwin/itwinjs-core#4284] [Ideas]
We have an interest in being able to both edit properties and add new properties to existing elements from our viewer that uses the iTwin frontend API.  We would also expect these properties to be intelligently managed upon subsequent iModel synchronizations.  That is, we don't necessarily want property values that the user has carefully modified or created to be lost by a subsequent iModel synchronization.  (Perhaps make it an iModel/Project configuration?)

**Comment by kabentley** [comment]
You can add new models to an existing iModel, and they won't be affected by synchronization. You can also add new `ElementAspects` to an element in a model that is synchronized, and they won't be affected by future synchronizations (although they'll be deleted if you delete their element). You can also usually add `JsonProperties` to elements without causing conflicts, since most connectors don't add them.

However, all changes to an iModel happen from the backend. When you say "from the frontend API", you'll have to be prepared to connect to a writeable backend.
